### PR TITLE
Fix 1654 - Change Dialog to use EDS

### LINKED ISSUE #1654: Change Dialog to EDS
**Is your feature request related to a problem? Please describe.**
Today the ModalDialog uses Material UI and the loading indicator could be better. 
Components should be changed to use EDS instead of Material UI

**Describe the solution you'd like**
Replace Material UI components with EDS components and change the loading indicator (see context image)

**Describe alternatives you've considered**
- Change to EDS
- Change loading indicator to [dots inside button](https://eds-storybook-react.azurewebsites.net/?path=/docs/feedback-progress-indicators-dots--introduction#inside-button)

**Additional context**
image

### PR DESCRIPTION
**Pull request by KindCoder-no**
## Fixes
This pull request fixes #1654 

## Description
Changed Dialog and Buttons to use EDS components

## Type of change
* Enhancement of existing functionality

## Impacted Areas in Application
- Frontend

## Checklist:
_Please tick all the boxes or remove the ones that aren't needed and explain why_

*Communication*
* [X] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [X] Code follows the style guidelines
* [X] I have self-reviewed my code
* [X] No new warnings are generated

*Test coverage*
* [X] Existing tests pass
* [ ] New code is covered by passing tests

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/Modals/ModalDialog.tsx [hunk 1]
```diff
@@ -70,67 +69,75 @@ const ModalDialog = (props: ModalDialogProps): React.ReactElement => {
...
         {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
-        <DialogActions>
-          {buttons[switchButtonPlaces ? 1 : 0]}
-          {buttons[switchButtonPlaces ? 0 : 1]}
-          {isLoading && <CircularProgress size="1.5rem" />}
-          {onDelete && buttons[2]}
-        </DialogActions>
       </Content>
+      <Dialog.Actions style={{ width: "100%" }}>
+        {isLoading ? (
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/Modals/ModalDialog.tsx]: Would it be better to do the isLoading check on line 62 to ensure that it is always the "confirm" buttan that turns to loading?

File: Src/WitsmlExplorer.Frontend/components/Modals/ServerModal.tsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ const ServerModal = (props: ServerModalProps): React.ReactElement => {
       <ModalDialog
         heading={`Remove the server "${server.name}"?`}
         content={<>Removing a server will permanently remove it from the list.</>}
-        confirmColor={"secondary"}
+        confirmColor={"danger"}
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/Modals/ServerModal.tsx:97]: There are multiple other modals that used "secondary" and now should use "danger" to match.
image

File: Src/WitsmlExplorer.Frontend/components/Modals/ModalDialog.tsx [hunk 3]
```diff
@@ -70,67 +69,75 @@ const ModalDialog = (props: ModalDialogProps): React.ReactElement => {
...
+          <StyledButton>
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/Modals/ModalDialog.tsx]: I think the button with progress dots should have style={ cursor:"not-allowed" } to indicate that it is not clickable if we are not going to use set it to disabled to preserve the style

File: Src/WitsmlExplorer.Frontend/components/Modals/ModalDialog.tsx (hunk 1 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/components/Modals/ModalDialog.tsx]: Fixed

File: Src/WitsmlExplorer.Frontend/components/Modals/ServerModal.tsx (hunk 2 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/components/Modals/ServerModal.tsx:97]: Fixed. (I also had to change allowed colors in ConfirmDialog)

File: Src/WitsmlExplorer.Frontend/components/Modals/ModalDialog.tsx (hunk 3 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/components/Modals/ModalDialog.tsx]: I don't think MUI / EDS supports changing of cursor

Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/components/Modals/ModalDialog.tsx]: Fixed

**Review by steinsiv — APPROVED** [review]
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Review by janburak — APPROVED** [review]
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**